src/darwin/Framework/CHIP/MTRDeviceController.h

```diff
@@ -85,9 +85,10 @@ typedef void (^MTRDeviceConnectionCallback)(MTRBaseDevice * _Nullable device, NS
  * after that point if it wants to commission the device.
  */
 - (BOOL)pairDevice:(uint64_t)deviceID
-     discriminator:(uint16_t)discriminator
```

Comment:
This API cannot be changed anymore.

In any case, setupCommissioningSessionWithPayload exists and is what you should be using.  All the pairDevice APIs in MTRDeviceController are deprecated.

Reply:
Thanks, i didn't realize it was deprecated. I added a deprecated tag to the pairDevice APIs and removed these changes in the MTRDeviceController code.